### PR TITLE
add rng parameter for measure

### PR DESCRIPTION
**Pull request by GiggleLiu**
fix https://github.com/QuantumBFS/Yao.jl/issues/194

### CODE REVIEW

File: src/measure.jl [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
     res = Matrix{Int}(undef, nshots, B)
     for ib=1:B
-        @inbounds res[:,ib] = _measure(view(pl,:,ib), nshots)
+        @inbounds res[:,ib] = _measure(rng, view(pl,:,ib), nshots)
     end
     return res
 end
 
-YaoBase.measure(::ComputationalBasis, reg::ArrayReg{1}, ::AllLocs; nshots::Int=1) = _measure(reg |> probs, nshots)
+YaoBase.measure(rng::AbstractRNG, ::ComputationalBasis, reg::ArrayReg{1}, ::AllLocs; nshots::Int=1) = _measure(rng, reg |> probs, nshots)
```
Comment (Roger-luo) [inline src/measure.jl]: I think you need to put a default RNG (MT RNG) there, or it breaks with previous version.

Comment (Roger-luo) [reply src/measure.jl]: CI should just pass, if you add a default RNG.

Comment (GiggleLiu) [reply src/measure.jl]: Tests passed locally. So what is the problem.

Comment (Roger-luo) [reply src/measure.jl]: Oh, OK. I think I need to merge YaoBase first.

**Pull request closed without merging**